org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.tests/src/org/eclipse/xtext/example/domainmodel/tests/ValidationTests.xtend

```diff
@@ -184,6 +186,54 @@ class ValidationTests {
 		]
 	}
 
+	@Test def void testDuplicatedOperation() {
+		val model = '''
+			entity E {
+				op m() {}
+				op m() {}
+			}
+		'''
+		model.parse => [
+			assertNumberOfIssues(2)
+			assertError(OPERATION, DUPLICATE_OPERATION, model.indexOf("m"), 1, "Duplicate operation m")
+			assertError(OPERATION, DUPLICATE_OPERATION, model.lastIndexOf("m"), 1, "Duplicate operation m")
+		]
+	}
+
+	@Test def void testDuplicatedOperationWithDifferentSignatureIsAllowed() {
+		val model = '''
+			entity E {
+				op m(int i) {}
+				op m() {}
+			}
+		'''
+		model.parse.assertNoErrors
```

Comment:
I prefer not to use the extra local variable `model` if it is used only in once place, but rather inline it:
```

'''
	entity E {
		op m(int i) {}
		op m() {}
	}
'''.parse.assertNoErrors
```

Reply:
OK, will do!